src/mame/machine/archimedes_keyb.h

```diff
@@ -0,0 +1,71 @@
+// license:LGPL-2.1+
+// copyright-holders:Angelo Salese, Sandro Ronco
+/**********************************************************************
+
+    Acorn Archimedes keyboard HLE
+
+*********************************************************************/
+
+#ifndef MAME_MACHINE_ARCHIMEDES_KEYB_H
+#define MAME_MACHINE_ARCHIMEDES_KEYB_H
+
+#pragma once
+
+#include "diserial.h"
+
+//**************************************************************************
+//  TYPE DEFINITIONS
+//**************************************************************************
+
+// ======================> archimedes_keyboard_device
+
+class archimedes_keyboard_device : public device_t, public device_serial_interface
```

Comment:
Any specific reason for not using `device_matrix_keyboard_interface` to handle scanning the key matrix?

Reply:
Every day is a good day to learn something new :)